pkg/checkpoint/protocol.go

```diff
@@ -41,81 +36,29 @@ const (
 	DefaultResendPeriod = types.Duration(time.Second)
 )
 
-// Protocol represents the state associated with a single instance of the checkpoint protocol
-// (establishing a single stable checkpoint).
-type Protocol struct {
-	logging.Logger
-
-	// IDs of modules the checkpoint tracker interacts with.
-	// TODO: Eventually put the checkpoint tracker in a separate package and create its own ModuleConfig type.
-	moduleConfig *ModuleConfig
-
-	// The ID of the node executing this instance of the protocol.
-	ownID t.NodeID
-
-	// Epoch to which this checkpoint belongs.
-	// It is always the epoch the checkpoint's associated sequence number (seqNr) is part of.
-	epoch tt.EpochNr
-
-	// Sequence number associated with this checkpoint protocol instance.
-	// This checkpoint encompasses seqNr sequence numbers,
-	// i.e., seqNr is the first sequence number *not* encompassed by this checkpoint.
-	// One can imagine that the checkpoint represents the state of the system just before seqNr,
-	// i.e., "between" seqNr-1 and seqNr.
-	seqNr tt.SeqNr
-
-	// The IDs of nodes to execute this instance of the checkpoint protocol.
-	// Note that it is the membership of epoch e-1 that constructs the membership for epoch e.
-	// (As the starting checkpoint for e is the "finishing" checkpoint for e-1.)
-	membership []t.NodeID
-
-	// State snapshot associated with this checkpoint.
-	stateSnapshot *commonpbtypes.StateSnapshot
-
-	// Hash of the state snapshot data associated with this checkpoint.
-	stateSnapshotHash []byte
-
-	// Set of (potentially invalid) nodes' signatures.
-	signatures map[t.NodeID][]byte
-
-	// Set of nodes from which a valid Checkpoint messages has been received.
-	confirmations map[t.NodeID]struct{}
-
-	// Set of Checkpoint messages that were received ahead of time.
-	pendingMessages map[t.NodeID]*checkpointpb.Checkpoint
-
-	// Time interval for repeated retransmission of checkpoint messages.
-	resendPeriod types.Duration
-
-	// Flag ensuring that the stable checkpoint is only announced once.
-	// Set to true when announcing a stable checkpoint for the first time.
-	// When true, stable checkpoints are not announced anymore.
-	announced bool
-}
-
-// NewProtocol allocates and returns a new instance of the Protocol associated with sequence number sn.
-func NewProtocol(
-	moduleConfig *ModuleConfig,
+// NewModule allocates and returns a new instance of the ModuleParams associated with sequence number sn.
+func NewModule(
+	moduleConfig *common.ModuleConfig,
 	ownID t.NodeID,
 	membership *commonpbtypes.Membership,
 	epochConfig *commonpbtypes.EpochConfig,
 	leaderPolicyData []byte,
 	resendPeriod types.Duration,
```

Comment:
I think it would be nicer to pack the module params into the `ModuleParams` struct and pass that as an argument to `NewModule`. This is how it's also done with most other modules. I'd do that for all those parameters except for the logger. I'd leave that one separate, as the intention is also to be able to eventually serialize the params. Also, conceptually, the logger is not really a parameter to a checkpoint protocol instance in the theoretical sense, just to this concrete implementation.

Reply:
Agreed, though it turned out to require a bit more refactoring than expected. (See upcoming push)